bioamptool.py

```diff
@@ -68,32 +70,53 @@
+            ser.write(b'WHORU\n') # Check board type
+            response = ser.readline().strip().decode()  # Try reading from the port
+            retry_counter += 1
+            if response in supported_boards:  # If response is received, assume it's the Arduino
+                global board
+                board = response # Set board type
+                print(f"{response} detected at {port}")  # Notify the user
+                if ser is not None:
+                    return ser  # Return the port name
+        ser.close()  # Close the port if no response
+    except (OSError, serial.SerialException):  # Handle exceptions if the port can't be opened
+        pass
+    print("Unable to connect to any hardware!")  # Notify if no Arduino is found
+    return None  # Return None if not found
+
 # Function to automatically detect the Arduino's serial port
-def auto_detect_arduino(baudrate, timeout=1):
+def detect_hardware(baudrate, timeout=1):
     ports = serial.tools.list_ports.comports()  # List available serial ports
+    ser = None
     for port in ports:  # Iterate through each port
-        try:
-            ser = serial.Serial(port.device, baudrate=baudrate, timeout=timeout)  # Try opening the port
-            ser.write(b'WHORU\n')
-            time.sleep(1)  # Wait for the device to initialize
-            response = ser.readline().strip().decode()  # Try reading from the port
-            if response:  # If response is received, assume it's the Arduino
-                global board
-                ser.close()  # Close the serial connection
-                print(f"{response} detected at {port.device}")  # Notify the user
-                board = response
-                return port.device  # Return the port name
-            ser.close()  # Close the port if no response
-        except (OSError, serial.SerialException):  # Handle exceptions if the port can't be opened
-            pass
-    print("Arduino not detected")  # Notify if no Arduino is found
+        ser = connect_hardware(port.device, baudrate)
+        if ser is not None:
+            return ser
+    print("Unable to detect hardware!")  # Notify if no Arduino is found
     return None  # Return None if not found
 
+def send_command(ser, command):
+    ser.flushInput()   # Clear the input buffer
+    ser.flushOutput()  # Clear the output buffer
+    ser.write((command + '\n').encode())  # Send command
+    time.sleep(0.1)  # Wait briefly to ensure Arduino processes the command
+    response = ser.readline().decode('utf-8', errors='ignore').strip()  # Read response
+    return response
+
 # Function to read data from Arduino
 def read_arduino_data(ser, csv_writer=None):
     global total_packet_count, cumulative_packet_count, previous_sample_number, missing_samples, buffer, data
-
     raw_data = ser.read(ser.in_waiting or 1)  # Read available data from the serial port
+    if raw_data == b'':
+        print("Raw Data:", raw_data)
```

Comment:
@PayalLakra we should not keep the debug print statement in the final code.

Reply:
Removed extra print statement.